src/web/mjs/connectors/NineAnime.mjs

```diff
@@ -139,23 +139,15 @@ export default class NineAnime extends Connector {
             new Promise((resolve, reject) => {
                 const timer = setInterval(() => {
                     try {
-                        if(button = document.querySelector('ul.episodes li a.active')) {
+                        const iframe = document.querySelector('div#player iframe')
+
+                        if (iframe) {
                             clearInterval(timer);
                             localStorage.setItem('player_autoplay', 0);
                             localStorage.setItem('_lastServerId', ${chapterID.server});
-                            new MutationObserver(mutations => {
-                                for(const mutation of mutations) {
-                                    for(const node of mutation.addedNodes) {
-                                        if(/iframe/i.test(node.tagName)) {
-                                            node.parentNode.removeChild(node);
-                                            resolve(node.src);
-                                        }
-                                    }
-                                }
-                            }).observe(document.querySelector('div#player'), {
-                                childList: true
-                            });
-                            button.click();
+                            resolve(iframe.src)
+                        } else {
+                            window.location.reload()
```

Comment:
I think that with the browser reload proposal  you're relying on the browser cache "fast load" but it would behave differently on potato computers and it would be a race condition.
The mutation observer needs to be fixed or altered.

Reply:
9anime may have overwritten the original click event, or are checking if the event arguments are from a valid mouse click...
There are other ways to raise the event programatically with corresponding mouse arguments as well

Reply:
We can make the click works with something like that : 

```javascript
  const clickEvent = new MouseEvent("click", {
      pointerX: getOffset(button).left,
      pointerY: getOffset(button).top,
      button: 0,
      ctrlKey: false,
      altKey: false,
      shiftKey: false,
      metaKey: false,
      bubbles: true,
      cancelable: false,
      relatedTarget: button
  });
  button.dispatchEvent(clickEvent)
```

The issue is that a "real" click return an object with a key `isTrusted` which got the value `true`. I still can't figure out how to mock this, I tried many things. When the click event return `{isTrusted: false, ...}` it reloads the page for some reason, so the `MutationObserver` is useless.

Reply:
In that case we could try to go full out and attach the chrome debugger to fire the mouse click (like done in JapScan)
https://stackoverflow.com/questions/34853588/how-to-trigger-an-istrusted-true-click-event-using-javascript